smithy-aws-protocol-tests/model/restJson1WithRenames/main.smithy

```diff
@@ -0,0 +1,34 @@
+$version: "1.0"
+
+namespace aws.protocoltests.restjsonWithRenames
+
+use aws.api#service
+use aws.protocols#restJson1
+use smithy.test#httpRequestTests
+use smithy.test#httpResponseTests
+
+/// A REST JSON service that sends JSON requests and responses.
+/// This test service is intended to test the `rename`
+/// functionality of services and it effect on codegen.
+@service(sdkId: "Rest Json Protocol")
+@restJson1
+service RestJsonWithRenames {
+    version: "2019-12-16",
+    rename: {
+        "aws.protocoltests.restjsonWithRenames.nested#Foo": "NestedFoo",
+    },
+    operations: [
+        SayHello,
+    ]
+}
+
+operation SayHello {
```

Comment:
I used a separate service to isolate the test case. Putting this on the normal rest-json service to me, implies it's a feature of rest-json, but really it's a generic codegen feature.

Reply:
True, and you're probably right that it could just be added to the rest-json tests.

Reply:
Having a separate service makes it easier to disable / ignore for code generators who haven't added the functionality yet (especially since this will probably prevent clients from compiling)